Filerename feature bugs

TrixiEther opened the pull request:
1. Fixed #116 
2. Fixed #123 - based on Windows OS specific prohibited characters
3. Added a check for a filename size limit of 255 characters, because the previous implementation did not work

TrixiEther committed:
TrixiEther | Filename attachment bugs

Done

N3-M3-S1S commented on src/com/mishiranu/dashchan/ui/posting/dialog/AttachmentOptionsDialog.java:
```diff
@@ -44,6 +44,8 @@ public class AttachmentOptionsDialog extends DialogFragment implements AdapterVi
 	public static final String TAG = AttachmentOptionsDialog.class.getName();
 
 	private static final String EXTRA_ATTACHMENT_INDEX = "attachmentIndex";
+	private static final String FILENAME_BLOCKED_CHARACTERS = "\\/:*?\"<>|.";
+	private static final int FILENAME_MAX_CHARACTER_COUNT = 255;
```
```suggestion
	private static final int FILENAME_MAX_CHARACTER_COUNT = 255;
```

N3-M3-S1S commented:
@TrixiEther As for now, if a user types a blocked character - all filename gets deleted. I don't know if this is a bug or a feature, but it may not be a good user experience. I'd like to suggest  different implementation of the filter. Should i do it here or crate a new PR?

TrixiEther commented:
@N3-M3-S1S it's weird because i just tested it. If the user enters a character, then it is simply not entered, the file name is not erased from me
Test environment: Android 11.0 Pixel 2 API 30

N3-M3-S1S commented:
@TrixiEther Here's how it looks like on Nexus 5 Android 11.
Seems like it doesn't reset if you try to insert a forbidden character into the original filename though.

[filename_reset.webm](https://github.com/TrixiEther/DashchanFork/assets/16612598/875cbd03-b02a-4e62-a2f6-5dfc2d7b3365)

N3-M3-S1S commented:
@TrixiEther Also, in media preferences you can enter forbidden symbols in "New filename" field.

TrixiEther commented:
@N3-M3-S1S interesting, I'll check it out when I get a chance

TrixiEther committed:
TrixiEther | Filename interface bugs

TrixiEther commented:
Updated changes. The functionality of EditPreference has been slightly extended with the ability to add custom filters for editText. Simplified filter by filename length by adding LenghtFilter. Constants moved to utility class for ease of use
